dependency/consul_partitions.go

```diff
@@ -0,0 +1,112 @@
+package dependency
+
+import (
+	"context"
+	"log"
+	"net/url"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/pkg/errors"
+)
+
+var (
+	// Ensure implements
+	_ Dependency = (*ListPartitionsQuery)(nil)
+
+	// ListPartitionsQuerySleepTime is the amount of time to sleep between
+	// queries, since the endpoint does not support blocking queries.
+	ListPartitionsQuerySleepTime = 15 * time.Second
```

Comment:
[NIT] wonder if we want to create just one `LongPollingSleepTime` set to 15 seconds for use across all of the non blocking query endpoints?

Reply:
so we kinda can, we do some re-assigning of these within tests and I'm not 100% on how that will play out if these tests run in parallel (since we could get a concurrent read and write) what I did though was move this to a `DefaultNonBlockingQuerySleepTime` constant and assign the individual sleep times to that const